Frameworks/GLKit/GLKMath.mm

```diff
@@ -919,18 +955,22 @@ GLKIT_EXPORT GLKMatrix3 GLKMatrix3MakeZRotation(float rad) {
  @Status Interoperable
 */
 GLKIT_EXPORT GLKMatrix4 GLKMatrix4MakeRotation(float rad, float x, float y, float z) {
-    float magn = sqrtf(x * x + y * y + z * z);
+    const float magn = sqrtf(x * x + y * y + z * z);
+    GLKMatrix4 res = { 0 };
+    float invMagn;
+
     if (magn < COMPARISON_EPSILON) {
-        return GLKMatrix4MakeIdentity();
+		// Detect near zero magnitude vector and set the inverse to NaN value to avoid divide by zero situation 
+		// and return the same output as iOS does in this case.
```

Comment:
I apologize for this. I had re-installed VS on another machine before editing and hadn't set it to use spaces instead of tabs yet nor used the clang formatter. It is done now.